docs/pretty-tables.md

```diff
@@ -0,0 +1,142 @@
+# `table.pretty` and changes to printing tables
+
+## Summary
+
+We will add a function to display a pretty version of a table for debugging purposes. We will use this function as the backbone to changing the default implementation of `print(table)` in CLI contexts.
+
+## Motivation
+
+In Roblox Studio, printing out a table with "Log mode" off will display an interactive table. In roblox-cli, Lune, and luau.exe, it will print the memory address, which is faintly useful, and makes debugging tables a massive chore. Making tables pretty print is a clear win in this regard.
+
+Adding a function to create these pretty forms and using that as the underlying implementation behind `print` is useful for custom loggers that do not use print and go to file, or for something like debug commands.
+
+## Design
+
+### `table.pretty`
+
+The following function will be added to the standard library:
+
+`table.pretty(input: { [any]: any }, indentation: number?): string`
+
+`indentation` defaults to 2.
+
+For a non-empty table, this will perform the following behavior:
+
+1. The initial opening and ending brace will be on their own line
+2. Form the string `[$key]: $value,` for every key and value, based on the same behavior as generalized iteration, even using `__iter` if it exists.
+	- `$key` and `$value` will be pretty printed if it is both a table and if it would be acceptable to do so (no `__metatable`). This will be indented `indentation` number of spaces.
+	- If `$value` is a string, it will be wrapped in quotes.
+	- Otherwise, `$key` and `$value` will be the result of `tostring(value)`.
+	- The comma is always included.
+	- `$key` is not included if it is a number part of a sequence of keys starting with `1`.
+3. In the event of a cyclic table with no tostring implementation, the literal string `"<cyclic: memory address>"` will be put in its place.
+
+If indentation is 0, then it will all be put onto a single line, with spaces being used in place of new lines. In this case, trailing commas will not be present.
+
+#### Examples
+
+```lua
+print(table.pretty({
+	width = 12,
```

Comment:
Nit: Need to change the indentation here or in the other Lua things. GitHub web editor just stuck to two spaces once I started using it elsewhere =P